server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -603,7 +603,7 @@ public void writeMapWithConsistentOrder(@Nullable Map<String, ? extends Object>
      * @param keyWriter The key writer
      * @param valueWriter The value writer
      */
-    public final <K, V> void writeMap(final ImmutableOpenMap<K, V> map, final Writer<K> keyWriter, final Writer<V> valueWriter)
+    public final <K, V> void writeImmutableMap(final ImmutableOpenMap<K, V> map, final Writer<K> keyWriter, final Writer<V> valueWriter)
```

Comment:
Do we really need this? Is there a way we can use the regular `writeMap` for writing these now that `ImmutableOpenMap` is an actual `Map`?

Reply:
I only renamed the method, to avoid needing to force casts. I agree that we should be able to get rid of it. However, there seems to be some path through it which results in a slightly different binary format for an ImmutableOpenMap vs a Map (in a previous commit writing an empty ImmutableOpenMap to a stream for snapshots caused tons of transport deserialization failures). We'll have to look at that in a followup.

Reply:
> However, there seems to be some path through it which results in a slightly different binary format

I ran into this when I tried it, now that we have more time hopefully we can figure it out :)